lib/note-content-editor.jsx

```diff
@@ -1,11 +1,68 @@
 import React, { PropTypes } from 'react';
-import { Editor, EditorState, ContentState } from 'draft-js';
+import {
+	ContentState,
+	Editor,
+	EditorState,
+	Modifier,
+} from 'draft-js';
 import { invoke, noop } from 'lodash';
 
 function plainTextContent( editorState ) {
 	return editorState.getCurrentContent().getPlainText( '\n' )
 }
 
+// contains methods that mirror the methods on Modifier with 2 differences:
+// 1. operate on EditorState instances
+// 2. move selection to the natural place following the modification
+const EditorStateModifier = {
+	insertText( editorState, selection, characters ) {
+		// this function assumes selection is collapsed
+
+		const afterInsert = EditorState.push(
+			editorState,
+			Modifier.insertText(
+				editorState.getCurrentContent(), selection, characters
+			),
+			'insert-characters'
+		);
+
+		const prevSelection = editorState.getSelection();
+		const prevOffset = prevSelection.getStartOffset();
+		const offsetDiff = characters.length;
+
+		// move selection forward by nr of chars that were added
+		const nextSelection = prevSelection.merge( {
+			anchorOffset: prevOffset + offsetDiff,
+			focusOffset: prevOffset + offsetDiff,
+		} )
+
+		return EditorState.forceSelection( afterInsert, nextSelection );
+	},
+
+	removeRange( editorState, selection = editorState.getSelection() ) {
```

Comment:
when I did my first couple passes I missed this, but I'm suspicious of the default value for `selection` here

I _do_ like using proper defaults on parameters, but just as with destructuring objects, it's easy to jump a little too far too quickly. in this case, we've got an input arg dependent on the arg passed in (or not passed in) _and_ dependent on another input arg passed in. would make for some odd debugging at best when something goes wrong.

constant values used as defaults: great
function calls as defaults: okay sometimes
dynamic, dependent, side-effecting actions as defaults: :shiver: